src/main/kotlin/app/revanced/patches/ad/VideoAdsPatch.kt

```diff
@@ -0,0 +1,103 @@
+package app.revanced.patches.ad
+
+import app.revanced.patcher.PatcherData
+import app.revanced.patcher.extensions.AccessFlagExtensions.Companion.or
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.*
+import app.revanced.patcher.signature.MethodMetadata
+import app.revanced.patcher.signature.MethodSignature
+import app.revanced.patcher.signature.MethodSignatureMetadata
+import app.revanced.patcher.signature.PatternScanMethod
+import app.revanced.patcher.smali.asInstructions
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+
+private val compatiblePackages = arrayOf("com.google.android.youtube")
+
+class VideoAdsPatch : Patch(
+    metadata = PatchMetadata(
+        shortName = "video-ads",
+        name = "YouTube Video Ads Patch",
+        description = "Patch to remove ads in the YouTube video player.",
+        compatiblePackages = compatiblePackages,
+        version = "0.0.1"
+    ),
+    signatures = listOf(
+        MethodSignature(
+            methodSignatureMetadata = MethodSignatureMetadata(
+                name = "show-video-ads-constructor",
+                methodMetadata = MethodMetadata(
+                    definingClass = "zai",
+                    name = "<init>",
+                ),
+                patternScanMethod = PatternScanMethod.Fuzzy(2),// FIXME: Test this threshold and find the best value.
+                compatiblePackages = compatiblePackages,
+                description = """Signature for the constructor of some class.
+                This signature is being used to find another method in the parent class
+                and was discovered in the YouTube version v17.03.38""",
+                version = "0.0.1"
```

Comment:
This is not what I initially meant with the `version` parameter for `MethodSignature`.
Even though this is fine, we now don't have any context for `methodMetadata`. How should we know which version that metadata is made for?

Reply:
The `version` parameter is just metadata for the user. So later it will be displayed in the manager and should be used to version signatures or patches. This way someone can down or upgrade patches and signatures. The `description` parameter or an additional one can be used to tell, which base version a signature and thus `MethodMetadata` has been generated for.